src/domain/mongoose_domain_sup.erl

```diff
@@ -0,0 +1,17 @@
+-module(mongoose_domain_sup).
+-export([start_link/0]).
+-export([init/1]).
+
+-ignore_xref([start_link/0]).
+
+-spec(start_link() ->
+    {ok, Pid :: pid()} | ignore | {error, Reason :: term()}).
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
+    SupFlags = #{strategy => rest_for_one,
+                 intensity => 100, period => 5,
+                 shutdown => 1000},
+    ChildSpecs = [],
```

Comment:
If we have here this supervisor, why don't we give the supervisor its child specs and we get the children started here, instead of hardcoded in the mongooseim:start() entry point? Also, a strategy of rest_for_one would restart all chlidren started _after_ the one that died (which kinda seems right here), and right now it is not clear in a single place the order that children are started as.